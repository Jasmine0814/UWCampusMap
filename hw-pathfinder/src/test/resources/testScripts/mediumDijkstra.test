# find the shortest path with Dijkstra's algorithm

CreateGraph Graph1

AddNode Graph1 A
AddNode Graph1 B
AddNode Graph1 C
AddNode Graph1 D
AddNode Graph1 E

AddEdge Graph1 B C 10.0
AddEdge Graph1 C A 1000.01
AddEdge Graph1 D E 750.0
AddEdge Graph1 E A 10.0
AddEdge Graph1 B D 20.0

FindPath Graph1 B A

