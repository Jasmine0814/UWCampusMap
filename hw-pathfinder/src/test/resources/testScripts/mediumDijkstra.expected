# find the shortest path with Dijkstra's algorithm

created graph Graph1

added node A to Graph1
added node B to Graph1
added node C to Graph1
added node D to Graph1
added node E to Graph1

added edge 10.000 from B to C in Graph1
added edge 1000.010 from C to A in Graph1
added edge 750.000 from D to E in Graph1
added edge 10.000 from E to A in Graph1
added edge 20.000 from B to D in Graph1

path from B to A:
B to D with weight 20.000
D to E with weight 750.000
E to A with weight 10.000
total cost: 780.000